cylc/flow/network/graphql.py

```diff
@@ -95,8 +95,13 @@ def instantiate_middleware(middlewares):
 def null_setter(result):
     """Set type to null if result is empty/null-like."""
     # Only set empty parents to null.
-    if result in EMPTY_VALUES:
-        return NULL_VALUE
+    try:
+        if result in EMPTY_VALUES:
+            return NULL_VALUE
+    except TypeError:
+        # If field is a repeated composite field convert to list.
+        if list(result) in EMPTY_VALUES:
+            return NULL_VALUE
```

Comment:
Does a "repeated composite field" not have a "empty value" you could add to `EMPTY_VALUES` to avoid this except-block and list conversion?

Reply:
Not sure one repeated composite field is equivalent to another (when empty)... 

Otherwise you'd still need to test that it's a protobuf object/field, and that it's a "repeated composite field" ... And since that type of field might not be too common, I thought it less expensive to just alternate on the exception.

Reply:
will check

Reply:
Sounds reasonable :+1:

Reply:
I'll approve and merge once you report back on the result of "will check"...

Reply:
Hmm..
```
>>> type(PbPrerequisite().conditions)
<class 'google.protobuf.pyext._message.RepeatedCompositeContainer'>
```
This works:
```
   assert PbTaskProxy().prerequisites == PREREQ.conditions
```
However, I don't think you can add it to the `EMPTY_VALUES` because if it's not empty it will be compared to the other types.. throwing the error..

Reply:
Ok I've removed the list conversion.. I think the exception can just test `not result`